diff --git a/eva.module b/eva.module
index 0d4b9699..69f2223e 100644
--- a/eva.module
+++ b/eva.module
@@ -223,5 +223,7 @@ function template_preprocess_eva_display_entity_view(&$variables) {
   $variables['view_array']['#view_id'] = $view->storage->id();
   $variables['view_array']['#view_display_show_admin_links'] = $view->getShowAdminLinks();
   $variables['view_array']['#view_display_plugin_id'] = $display->getPluginId();
-  views_add_contextual_links($variables['view_array'], 'view', $display->getLinkDisplay());
+  if ($link_display = $display->getLinkDisplay()) {
+    views_add_contextual_links($variables['view_array'], 'view', $display->getLinkDisplay());
+  }
 }
