diff --git a/group_content_menu.module b/group_content_menu.module
index 79b72e7..a9cb17f 100644
--- a/group_content_menu.module
+++ b/group_content_menu.module
@@ -129,8 +129,8 @@ function group_content_menu_module_implements_alter(&$implementations, $hook) {
  * Implements hook_form_BASE_FORM_ID_alter().
  */
 function group_content_menu_form_node_form_alter(&$form, FormStateInterface $form_state) {
-  \Drupal::classResolver(NodeFormAlter::class)
-    ->alter($form, $form_state);
+  // \Drupal::classResolver(NodeFormAlter::class)
+  //   ->alter($form, $form_state);
 }
 
 /**
