From 387c5be3b89783e6d88843219c88b9c3a535bbf1 Mon Sep 17 00:00:00 2001
From: Oden <hello@odensc.com>
Date: Fri, 18 Nov 2022 11:07:12 -0800
Subject: [PATCH] Unset "none" option in state list

---
 scheduler_content_moderation_integration.module | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/scheduler_content_moderation_integration.module b/scheduler_content_moderation_integration.module
index e633fda..97e4d5f 100644
--- a/scheduler_content_moderation_integration.module
+++ b/scheduler_content_moderation_integration.module
@@ -146,6 +146,10 @@ function scheduler_content_moderation_integration_form_alter(&$form, FormStateIn
   // selection field.
   $form['publish_state']['#access'] = ($form['publish_state']['#access'] ?? TRUE) && $scheduler_manager->getThirdPartySetting($entity, 'publish_enable', $config->get('default_publish_enable'));
   $form['unpublish_state']['#access'] = ($form['unpublish_state']['#access'] ?? TRUE) && $scheduler_manager->getThirdPartySetting($entity, 'unpublish_enable', $config->get('default_unpublish_enable'));
+
+  // Unset none in options list so we default to the correct moderation state
+  unset($form['publish_state']['widget'][0]['#options']['_none']);
+  unset($form['unpublish_state']['widget'][0]['#options']['_none']);
 }

 /**
--
2.34.1

