From 8a7a14603f6091ae5b46e821d2aa4a72078db780 Mon Sep 17 00:00:00 2001
From: Stephen Mustgrave <smustgrave@gmail.com>
Date: Sat, 28 May 2022 12:24:18 -0400
Subject: [PATCH 1/6] Make fields conditional

---
 ...uler_content_moderation_integration.module | 22 +++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/scheduler_content_moderation_integration.module b/scheduler_content_moderation_integration.module
index 04ef1f7..d4f0e5c 100644
--- a/scheduler_content_moderation_integration.module
+++ b/scheduler_content_moderation_integration.module
@@ -133,6 +133,28 @@ function scheduler_content_moderation_integration_form_node_form_alter(&$form, F
   $type = $form_state->getFormObject()->getEntity()->type->entity;
   $form['publish_state']['#access'] = ($form['publish_state']['#access'] ?? TRUE) && $type->getThirdPartySetting('scheduler', 'publish_enable', $config->get('default_publish_enable'));
   $form['unpublish_state']['#access'] = ($form['unpublish_state']['#access'] ?? TRUE) && $type->getThirdPartySetting('scheduler', 'unpublish_enable', $config->get('default_unpublish_enable'));
+
+  $form['publish_state']['#states'] = [
+    'visible' => [
+      // Don't mistake :input for the type of field -- it's just a css selector.
+      // You can always use :input or any other css selector here, no matter
+      // whether your source is a select, radio or checkbox element.
+      ':input[name="publish_on[0][value][date]"]' => ['!value' => ''],
+      'and',
+      ':input[name="publish_on[0][value][time]"]' => ['!value' => ''],
+    ],
+  ];
+
+  $form['unpublish_state']['#states'] = [
+    'visible' => [
+      // Don't mistake :input for the type of field -- it's just a css selector.
+      // You can always use :input or any other css selector here, no matter
+      // whether your source is a select, radio or checkbox element.
+      ':input[name="unpublish_on[0][value][date]"]' => ['!value' => ''],
+      'and',
+      ':input[name="unpublish_on[0][value][time]"]' => ['!value' => ''],
+    ],
+  ];
 }

 /**
--
GitLab


From c12d727ec1899cbb38458bc9fe909c975d0ed4ab Mon Sep 17 00:00:00 2001
From: Stephen Mustgrave <smustgrave@gmail.com>
Date: Sun, 29 May 2022 13:14:36 -0400
Subject: [PATCH 2/6] Unset none in list

---
 scheduler_content_moderation_integration.module | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/scheduler_content_moderation_integration.module b/scheduler_content_moderation_integration.module
index d4f0e5c..13f4cfc 100644
--- a/scheduler_content_moderation_integration.module
+++ b/scheduler_content_moderation_integration.module
@@ -155,6 +155,9 @@ function scheduler_content_moderation_integration_form_node_form_alter(&$form, F
       ':input[name="unpublish_on[0][value][time]"]' => ['!value' => ''],
     ],
   ];
+
+  unset($form['publish_state']['widget'][0]['#options']['_none']);
+  unset($form['unpublish_state']['widget'][0]['#options']['_none']);
 }

 /**
--
GitLab


From 3342d0fae0a6deb32d0d43f4b5a0c5a4d4a27cf3 Mon Sep 17 00:00:00 2001
From: Stephen Mustgrave <smustgrave@gmail.com>
Date: Sun, 29 May 2022 17:29:09 -0400
Subject: [PATCH 3/6] Minor update

---
 ...uler_content_moderation_integration.module | 54 +++++++++----------
 1 file changed, 27 insertions(+), 27 deletions(-)

diff --git a/scheduler_content_moderation_integration.module b/scheduler_content_moderation_integration.module
index 13f4cfc..d55ca53 100644
--- a/scheduler_content_moderation_integration.module
+++ b/scheduler_content_moderation_integration.module
@@ -131,33 +131,33 @@ function scheduler_content_moderation_integration_form_node_form_alter(&$form, F
   $config = \Drupal::config('scheduler.settings');
   /** @var \Drupal\node\NodeTypeInterface $type */
   $type = $form_state->getFormObject()->getEntity()->type->entity;
-  $form['publish_state']['#access'] = ($form['publish_state']['#access'] ?? TRUE) && $type->getThirdPartySetting('scheduler', 'publish_enable', $config->get('default_publish_enable'));
-  $form['unpublish_state']['#access'] = ($form['unpublish_state']['#access'] ?? TRUE) && $type->getThirdPartySetting('scheduler', 'unpublish_enable', $config->get('default_unpublish_enable'));
-
-  $form['publish_state']['#states'] = [
-    'visible' => [
-      // Don't mistake :input for the type of field -- it's just a css selector.
-      // You can always use :input or any other css selector here, no matter
-      // whether your source is a select, radio or checkbox element.
-      ':input[name="publish_on[0][value][date]"]' => ['!value' => ''],
-      'and',
-      ':input[name="publish_on[0][value][time]"]' => ['!value' => ''],
-    ],
-  ];
-
-  $form['unpublish_state']['#states'] = [
-    'visible' => [
-      // Don't mistake :input for the type of field -- it's just a css selector.
-      // You can always use :input or any other css selector here, no matter
-      // whether your source is a select, radio or checkbox element.
-      ':input[name="unpublish_on[0][value][date]"]' => ['!value' => ''],
-      'and',
-      ':input[name="unpublish_on[0][value][time]"]' => ['!value' => ''],
-    ],
-  ];
-
-  unset($form['publish_state']['widget'][0]['#options']['_none']);
-  unset($form['unpublish_state']['widget'][0]['#options']['_none']);
+
+  $show_publish_state = ($form['publish_state']['#access'] ?? TRUE) && $type->getThirdPartySetting('scheduler', 'publish_enable', $config->get('default_publish_enable'));
+  $show_unpublish_state = ($form['unpublish_state']['#access'] ?? TRUE) && $type->getThirdPartySetting('scheduler', 'unpublish_enable', $config->get('default_unpublish_enable'));
+  $form['publish_state']['#access'] = $show_publish_state;
+  $form['unpublish_state']['#access'] = $show_unpublish_state;
+
+  if ($show_publish_state) {
+    $form['publish_state']['#states'] = [
+      'visible' => [
+        ':input[name="publish_on[0][value][date]"]' => ['!value' => ''],
+        'and',
+        ':input[name="publish_on[0][value][time]"]' => ['!value' => ''],
+      ],
+    ];
+    unset($form['publish_state']['widget'][0]['#options']['_none']);
+  }
+
+  if ($show_unpublish_state) {
+    $form['unpublish_state']['#states'] = [
+      'visible' => [
+        ':input[name="unpublish_on[0][value][date]"]' => ['!value' => ''],
+        'and',
+        ':input[name="unpublish_on[0][value][time]"]' => ['!value' => ''],
+      ],
+    ];
+    unset($form['unpublish_state']['widget'][0]['#options']['_none']);
+  }
 }

 /**
--
GitLab


From 4fdb579782253588e3cc7a8fb4af484099015db1 Mon Sep 17 00:00:00 2001
From: Christopher Lewis <christopher.lewis@brandcode.de>
Date: Tue, 14 Mar 2023 18:20:42 +0100
Subject: [PATCH 4/6] Avoid fatal error when editing entities without type
 property

---
 scheduler_content_moderation_integration.module | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/scheduler_content_moderation_integration.module b/scheduler_content_moderation_integration.module
index 5e5bdb9..4801b60 100644
--- a/scheduler_content_moderation_integration.module
+++ b/scheduler_content_moderation_integration.module
@@ -337,11 +337,11 @@ function scheduler_content_moderation_integration_form_alter(&$form, FormStateIn
   // corresponding state selection field. If #access is already set to false
   // (for example by a third-party module) then do not override that setting.
   $config = \Drupal::config('scheduler.settings');
-  /** @var \Drupal\node\NodeTypeInterface $type */
-  $type = $form_state->getFormObject()->getEntity()->type->entity;
+  /** @var \Drupal\Core\Entity\EntityInterface $entity */
+  $entity = $form_state->getFormObject()->getEntity();

-  $show_publish_state = ($form['publish_state']['#access'] ?? TRUE) && $type->getThirdPartySetting('scheduler', 'publish_enable', $config->get('default_publish_enable'));
-  $show_unpublish_state = ($form['unpublish_state']['#access'] ?? TRUE) && $type->getThirdPartySetting('scheduler', 'unpublish_enable', $config->get('default_unpublish_enable'));
+  $show_publish_state = ($form['publish_state']['#access'] ?? TRUE) && $scheduler_manager->getThirdPartySetting($entity, 'publish_enable', $config->get('default_publish_enable'));
+  $show_unpublish_state = ($form['unpublish_state']['#access'] ?? TRUE) && $scheduler_manager->getThirdPartySetting($entity, 'unpublish_enable', $config->get('default_unpublish_enable'));
   $form['publish_state']['#access'] = $show_publish_state;
   $form['unpublish_state']['#access'] = $show_unpublish_state;

--
GitLab


From faf53036edba7526a00044cc21fb1cc8b690e904 Mon Sep 17 00:00:00 2001
From: Jeroen Tubex <jeroen.tubex@intracto.com>
Date: Wed, 19 Apr 2023 12:28:56 +0200
Subject: [PATCH 5/6] Fix failing tests

---
 scheduler_content_moderation_integration.module | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scheduler_content_moderation_integration.module b/scheduler_content_moderation_integration.module
index 4801b60..74f9fa0 100644
--- a/scheduler_content_moderation_integration.module
+++ b/scheduler_content_moderation_integration.module
@@ -340,8 +340,8 @@ function scheduler_content_moderation_integration_form_alter(&$form, FormStateIn
   /** @var \Drupal\Core\Entity\EntityInterface $entity */
   $entity = $form_state->getFormObject()->getEntity();

-  $show_publish_state = ($form['publish_state']['#access'] ?? TRUE) && $scheduler_manager->getThirdPartySetting($entity, 'publish_enable', $config->get('default_publish_enable'));
-  $show_unpublish_state = ($form['unpublish_state']['#access'] ?? TRUE) && $scheduler_manager->getThirdPartySetting($entity, 'unpublish_enable', $config->get('default_unpublish_enable'));
+  $show_publish_state = isset($form['publish_state']['#type']) && ($form['publish_state']['#access'] ?? TRUE) && $scheduler_manager->getThirdPartySetting($entity, 'publish_enable', $config->get('default_publish_enable'));
+  $show_unpublish_state = isset($form['unpublish_state']['#type']) && ($form['unpublish_state']['#access'] ?? TRUE) && $scheduler_manager->getThirdPartySetting($entity, 'unpublish_enable', $config->get('default_unpublish_enable'));
   $form['publish_state']['#access'] = $show_publish_state;
   $form['unpublish_state']['#access'] = $show_unpublish_state;

--
GitLab


From 1a6e5b3dd374c0f984c80da37c86062c21a951ad Mon Sep 17 00:00:00 2001
From: Jonathan Smith <jonathan1055@sandfordsolutions.com>
Date: Sat, 22 Apr 2023 18:52:18 +0100
Subject: [PATCH 6/6] Fix missing key coding standard

---
 scheduler_content_moderation_integration.module | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/scheduler_content_moderation_integration.module b/scheduler_content_moderation_integration.module
index 74f9fa0..e413e75 100644
--- a/scheduler_content_moderation_integration.module
+++ b/scheduler_content_moderation_integration.module
@@ -347,9 +347,9 @@ function scheduler_content_moderation_integration_form_alter(&$form, FormStateIn

   if ($show_publish_state) {
     $form['publish_state']['#states'] = [
+      // The #states visible conditions are always 'AND'ed together.
       'visible' => [
         ':input[name="publish_on[0][value][date]"]' => ['!value' => ''],
-        'and',
         ':input[name="publish_on[0][value][time]"]' => ['!value' => ''],
       ],
     ];
@@ -360,7 +360,6 @@ function scheduler_content_moderation_integration_form_alter(&$form, FormStateIn
     $form['unpublish_state']['#states'] = [
       'visible' => [
         ':input[name="unpublish_on[0][value][date]"]' => ['!value' => ''],
-        'and',
         ':input[name="unpublish_on[0][value][time]"]' => ['!value' => ''],
       ],
     ];
--
GitLab
