diff --git a/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorage.php b/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorage.php
index 45b1c363..cea15ab5 100644
--- a/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorage.php
+++ b/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorage.php
@@ -840,7 +840,13 @@ protected function doDeleteFieldItems($entities) {
    */
   public function save(EntityInterface $entity) {
     try {
-      $transaction = $this->database->startTransaction();
+      try {
+        $transaction = $this->database->startTransaction();
+      }
+      catch (\Exception $e) {
+        \Drupal::logger('php')->log(RfcLogLevel::DEBUG, "SqlContentEntityStorage::save() - Attempting recovery from database exception: " . $e->getMessage());
+        $transaction = $this->database->startTransaction();
+      }
       $return = parent::save($entity);
       $transaction->commitOrRelease();
 
