--- a/modules/contrib/tour/js/tour.js
+++ b/modules/contrib/tour/js/tour.js
@@ -254,7 +254,7 @@ import 'shepherd.js/dist/css/shepherd.css';
                     );
 
                   if (nextButton && typeof nextButton.focus === 'function') {
-                    nextButton.focus();
+                    nextButton.focus({ preventScroll: true });
                   }
                 },
               };
